Documentation/hierarchical-3d-registration.md

```diff
@@ -0,0 +1,83 @@
+
+The Hierarchical 3D Registration module utilizes a robust image registration algorithm based on rigid transformations. The algorithm is inspired by the method proposed by [Marai et al. 2006](http://dx.doi.org/10.1109/TMI.2005.862151) and implemented using [ITKElastix](https://elastix.dev/index.php). The bone hierarchy defines the structure of objects to be registered, arranged to reflect the anatomical relationships between parent and child bones. The process starts by defining regions of interest (ROIs) around the bones in the static 3D CT image (Source Volume), which will be compared against the corresponding regions in each frame of the dynamic dataset (CT Sequence). The entire hierarchy is registered in each frame, traversed in a breadth-first manner  starting with the root bone. During the registration of a given bone from the source volume to a specific frame in the sequence, the module provides a initial transformation that serves as the starting point for the optimizer. Once the optimal transformation from the source region to the target region is found, it is propagated to all child bones in the hierarchy, adjusting the starting position for each subsequent bone in the current frame. This hierarchical approach enhances the optimization process by accounting for motion constraints and thereby improving the alignment accuracy throughout the sequence.
+
+## General Inputs and Outputs
+
+The Hierarchical 3D Registration module requires four inputs:
+
+* CT Sequence: A sequence node containing the serial CT volume images to be tracked
+* Source Volume: A static 3D CT image to register from, the scalar volume node from which the model hierarchy is generated
+* Model Hierarchy: The root node of the model hierarchy, representing the rigid objects to be registered
+* Frames: The range of frames to be tracked
+
+The output of the module is a sequence of transforms for each bone, mapping from the bone's pose in the source volume to its pose in each frame. The module also generates additional nodes in the scene used during the registration process, such as:
+* Region of interest (ROI) nodes, used to define the regions to compare from the source volume to each sequence frame
+* Cropped volumes based on the ROIs of the source volume and sequence frames
+
+## Preparing the Model Hierarchy
+
+1) Load in the STL Models previously segmented from the Source Volume. If not yet processed, see the
+[Pre-Processing Auto-Generated Segmentation](https://autoscoper.readthedocs.io/en/latest/tutorials/pre-processing-module.html#auto-generated-segmentations) SAM module.
```

Comment:
For internal references of pages within the docs built in this git repo, we can specify the path to the source markdown file relative to the `Documentation` directory in this repo, and if we wish to reference a specific section of that page, append `#` to it and then the section reference (which is like the text in the section title, but hyphenated and in lower case. Read more [here](https://mystmd.org/spec/references)). So in this case we could do something like 
```suggestion
[Pre-Processing Auto-Generated Segmentation](tutorials/pre-processing-module.md#auto-generated-segmentations) SAM module.
```